### DISCUSSION TITLE
mass-Install software

### DISCUSSION THREAD

**Discussion by OlexTratisky** [Velocidex/velociraptor#2262] [General]
Hi, 

Can Velociraptor be used to mass deploy software on the endppints? Something like Ansible?

Any pro/cons?

thanks

**Comment by OlexTratisky** [comment]
anyone please? Can i use it to deploy software,security patches etc? fir Windows and Linux machines ? replacing sccm and ansible?

**Comment by scudette** [comment]
You can deploy any binary by way of "tools" - here is documentation https://docs.velociraptor.app/docs/extending_vql/#using-external-tools

But this is really designed to support VQL artifacts (so e.g. a tool is pushed to run a particular collection). It is not really a proper software management tool like chef, puppet ansible etc. Those are fully featured and mature tools which are well understood by many professional system admins.  Although Velociraptor can serve as the basis for this functionality you would have to mostly build it out yourself and you will end up with something that is not well understood by sysadmin and is a very custom solution. I do not recommend using Velociraptor for this purpose right now. If many people feel that is a real use case then maybe this feature can be properly developed over time but we need to design and think about it.

I can certainly see the appeal of a single platform to manage all endpoint needs including software packages. Usually we rely on software distribution tools to deploy Velociraptor itself :-).
